### PR TITLE
[repo] Release process tweaks & improvements 4

### PR DESCRIPTION
**Pull request by CodeBlanch**
## Changes

* Adds a new workflow `prepare-release.yml` which automates/streamlines the following tasks in the release process:
  * Automatically open a PR to update CHANGELOGs and public api files (stable releases only) for the tag being released
  * Once that PR is merged (by a maintainer) you can interact with the workflow\bot and reply with a comment containing "/CreateReleaseTag" in the body to instruct the workflow\bot to create the tag and then kick off the package workflow which...
     * Creates packages as artifacts on the workflow
     * Creates draft release for when packates have been uploaded
     * Creates post-release props update PR (stable releases only)

## Example

Here is a test run of the process: https://github.com/CodeBlanch/opentelemetry-dotnet/pull/36

## Merge requirement checklist

* [X] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (license requirements, nullable enabled, static analysis, etc.)

### CODE REVIEW

**Comment by TimothyMothra** [comment]
Are there restrictions on who can comment with the "/CreateReleaseTag" ?

**Comment by CodeBlanch** [comment]
@TimothyMothra Good question! Yes. What I did is have the bot lock the PR. While locked only collaborators of the repo may comment which limits who can speak "/CreateReleaseTag" 🤣

image

The bot will ignore "/CreateReleaseTag" if the PR isn't locked when it is posted.

File: .github/workflows/prepare-release.yml [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+
+on:
+  workflow_dispatch:
+    inputs:
+      tag-prefix:
+        type: choice
+        options:
+        - core-
+        - coreunstable-
+        - Instrumentation.AspNetCore-
```
Comment (vishweshbankwar) [inline .github/workflows/prepare-release.yml:11]: We should be able to remove these now?

Comment (CodeBlanch) [reply .github/workflows/prepare-release.yml]: 👍 updated

File: .markdownlint.yaml [hunk 2]
```diff
@@ -1,7 +1,12 @@
...
-# allow long lines for tables and code blocks
+# MD013/line-length : Line length : https://github.com/DavidAnson/markdownlint/blob/v0.32.1/doc/md013.md
 MD013:
   code_blocks: false
   tables: false
+
+# MD033/no-inline-html : Inline HTML : https://github.com/DavidAnson/markdownlint/blob/v0.32.1/doc/md033.md
+MD033:
+  # Allowed elements
+  allowed_elements: [ 'details', 'summary' ]
```
Comment (reyang) [inline .markdownlint.yaml:12]:
```suggestion
  allowed_elements: [ 'details', 'summary' ]

```

**Review by reyang — APPROVED** [review]

**Review by vishweshbankwar — APPROVED** [review]

**Pull request merged**